zerver/lib/outgoing_webhook.py

```diff
@@ -189,20 +190,30 @@ def do_rest_call(rest_operation, request_data, event, service_handler, timeout=N
     final_url = urllib.parse.urljoin(rest_operation['base_url'], rest_operation['relative_url_path'])
     request_kwargs = rest_operation['request_kwargs']
     request_kwargs['timeout'] = timeout
-
     try:
         response = requests.request(http_method, final_url, data=request_data, **request_kwargs)
         if str(response.status_code).startswith('2'):
             response_message = service_handler.process_success(response, event)
             if response_message is not None:
                 succeed_with_message(event, response_message)
-
-        # On 50x errors, try retry
-        elif str(response.status_code).startswith('5'):
-            request_retry(event, "Internal Server error at third party.")
         else:
-            failure_message = "Third party responded with %d" % (response.status_code)
-            fail_with_message(event, failure_message)
+            context = zulip_default_context(request_data)
+            message_url = ("{server}/#narrow/stream/{stream}/subject/{subject}/near/{id}"
+                           .format(server=context['realm_uri'],
```

Comment:
I used .format because it seemed clearer to me in this context, but I understand that % would be more consistent and shorter here.

Reply:
You can use `%(server)s` to do the same thing with `%`.